pinot-core/src/main/java/org/apache/pinot/core/accounting/PerQueryCPUMemAccountantFactory.java

```diff
@@ -291,6 +326,39 @@ public boolean throttleQuerySubmission() {
       return getWatcherTask().getHeapUsageBytes() > getWatcherTask().getQueryMonitorConfig().getAlarmingLevel();
     }
 
+    public void checkMemoryAndInterruptIfExceeded() {
```

Comment:
Instead of interrupting the anchor thread, this function can be called from `Tracing.ThreadAccountantOps.isInterrupted()`. Specifically:

```
return Tracing.getThreadAccountant.checkMemoryAndInterruptIfExceeded() || Thread.interrupted() || Tracing.getThreadAccountant().isAnchorThreadInterrupted();
```


A better long term solution is to change ``Tracing.ThreadAccountantOps.isInterrupted()` to

```
return Thread.interrupted() || Tracing.getThreadAccountant().isQueryTerminated();
```

`ThreadResourceUsageAccountant.isQueryTerminated()` is a new function:

```
return checkMemoryAndInterruptIfExceeded() || isAnchorThreadInterrupted()
```

The advantage is that new conditions can be added depending on the capability of the accountant such as also interrupting in panic mode.